gap/base/methsel.gi

```diff
@@ -20,39 +20,45 @@
 # Add a method to a database with "AddMethod" and call a method from a
 # database with "CallMethods".
 #
-InstallGlobalFunction("AddMethod", function(methodDb, method)
-    local l, pos;
-    if not IsSubsetSet(NamesOfComponents(method),
-                       ["method", "rank", "stamp"]) then
-        ErrorNoReturn("<method> must have components",
-                      " method, rank, and stamp");
+InstallGlobalFunction("AddMethod", function(methodDb, args...)
+    local method, rank, comment, l, pos;
+    # HACK to be compatible with "old methods" during switch to RecogMethod
+    # objects.
+    method := args[1];
+    if Length(args) = 1 then
+        rank := method.rank;
+        if not IsBound(method.comment) then
+            comment := "";
+        else
+            comment := method.comment;
+        fi;
+        method := RecogMethod(method.stamp, comment, method.method);
+    elif Length(args) = 2 then
+        rank := args[2];
+    else
+        ErrorNoReturn("wrong nr arguments");
     fi;
-    if not IsBound(method.comment) then
-        method.comment := "";
+    # end of HACK
+    if not IsRecogMethod(method) then
+        ErrorNoReturn("<method> must be a RecogMethod");
     fi;
-    # method.validatesOrAlwaysValidInput for now only stores meta-information.
-    # It may be used in the future, see github issue #184.
-    if not IsBound(method.validatesOrAlwaysValidInput) then
-        Info(InfoAddMethod, 1, "AddMethod, ", method.stamp,
-             ": validatesOrAlwaysValidInput not bound");
+    # This may be turned into an error in the future, see github issue #184.
+    if not HasValidatesOrAlwaysValidInput(method) then
+        Info(InfoAddMethod, 1, "AddMethod, ", Stamp(method),
+             ": has no ValidatesOrAlwaysValidInput");
     fi;
     l := Length(methodDb);
-    pos := First([1 .. l], i -> methodDb[i].rank <= method.rank);
+    pos := First([1 .. l], i -> methodDb[i].rank <= rank);
     if pos = fail then
         pos := l + 1;
     fi;
```

Comment:
I think this also works (but didn't test, beware of off-by-one errors etc.)
```suggestion
    pos := PositionSortedBy(methodDb, -rank, m -> -m.rank);
```
and then the local variable `l` could be removed.

Reply:
I'm going to install a method for `\<` for `RecogMethod`s, unless you object. Then we can use `PositionSorted` etc.

Reply:
But why? We are comparing ranks, not recognition methods